js-core/homeworks/homework_6/src/main.js

```diff
@@ -0,0 +1,502 @@
+
+let user = {
+  name: 'Albina',
+};
+
+let javaScript = {
+  html: 'JavaScript',
+};
+
+function countLetterA(str) {
+  return str.split('').reduce(function(accumulator,value){
+    return accumulator + (value === 'a' ? 1 : 0);      
+  },0);
+}
+
+console.log('task 1:')
+console.log(randomString , '=>' ,countLetterA(randomString));
+console.log(user.name + javaScript.html , '=>' ,countLetterA(user.name + javaScript.html));
+// console.log(countLetterA(user.name + javaScript.html)); // 3
+
+// ! ! ! ! ! ! ! ! ! ! !
+/*
+ *
+ * TASK 2
+ * Вам дано предложение, вам необходимо перевернуть каждое
+ * слово в предложении в обратном порядке
+ *
+ * */
+
+function reverseEachWord(sentence, direction) {
+  let sentenceArr = sentence.split(' ');
+  // let revercedWords = []
+
+  let revercedWords = sentenceArr.map(function(currentValue,index,arr){    
+      return currentValue.split('').reverse().join('');
+  });
+
+  if(!direction){
+    return revercedWords.join(' ');    
+  }else{
+    return revercedWords.reverse().join(' ');    
+  }
+}
+
+console.log('\ntask 2:');
+console.log('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM' , ' => ', reverseEachWord('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM'));
+// uoY t'nod evah ot od gnihtyna laiceps ot nigeb gnisu eht .MOD tnereffiD sresworb evah tnereffid snoitatnemelpmi fo eht MOD
+
+console.log('The Document Object Model (DOM) is a programming interface for HTML and XML documents', '=> ', reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents'));
+// ehT tnemucoD tcejbO ledoM )MOD( si a gnimmargorp ecafretni rof LMTH dna LMX stnemucod
+
+/*
+ * TASK 3
+ * Добавьте в функцию reverseEachWord второй параметр,
+ * данный параметр булево, если true - тогда предложение так же
+ * переворачиваются в обратном порядке
+ * */
+console.log('\ntask 3:');
+
+
+console.log(reverseEachWord('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM', true));
+//MOD eht fo snoitatnemelpmi tnereffid evah sresworb tnereffiD .MOD eht gnisu nigeb ot laiceps gnihtyna od ot evah t'nod uoY
+console.log(reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents', true));
+// stnemucod LMX dna LMTH rof ecafretni gnimmargorp a si )MOD( ledoM tcejbO tnemucoD ehT
+
+console.log(reverseEachWord('Hi my Name is', false));
+// iH ym emaN si
+
+/* TASK 4
+ * Посчитайте сколько одинаковых слов в предложении.
+ * Функция должна возвращать объект у которого свойства
+ * это повторяющиеся слово,
+ * а значение это количество повторений
+ * */
+
+// Both - Java - and - Java - Script - is - programming - and - programming - OOPBased
+
+function wordCounter(sentence) {
+  let obj =  {};
+  sentence.split(' ').forEach(function(value){
```

Comment:
Have you thought about using ".reduce" in that solution?

It fit very good for tasks like this one